### DISCUSSION TITLE
I don't understand why my operation freezes/does not work.

### DISCUSSION THREAD

**Discussion by bcheevers123** [mitre/caldera#2643] [Q&A]
Hello, I am trying to run a modified version of the default Windows worm operation. I am trying to use the worm lateraly move across my network of two Windows 10 home VMs. The Windows VMs have no domain name/ just use IP address. This meant I could remove the nslookup part of the operation out.

I tried to remove the requirement for the remote.host.fqdn on the Copy 54ndc7 (SMB) ability by editing the command run, but it still reports that it requires this var.

It is assumed that a caldera shell has started in the first Windows machine.
[Windows_Worm_IPAddress_only.txt](https
[debrief_2022-08-16_19-20-11.txt](https://github.com/mitre/caldera/files/9353483/debrief_2022-08-16_19-20-11.txt)
://github.com/mitre/caldera/files/9353481/Windows_Worm_IPAddress_only.txt)

The advesary:
![image](https://user-images.githubusercontent.com/89908497/184953682-62a62e09-e82f-4cff-bd6c-4ab1b838f2c7.png)

The view remote share command works/is successful, returns:

![image](https://user-images.githubusercontent.com/89908497/184953816-c41a8c96-09bf-4cf7-896e-571ffd356114.png)

The fact sources:

![unknown](https://user-images.githubusercontent.com/89908497/184954001-9bf44a74-8062-4ce1-9ac4-4d3f89669e97.png)

The command that either does not get executed, or gets stuck:

![unknown](https://user-images.githubusercontent.com/89908497/184954183-307e4bb6-82fc-46cc-9e97-2ef6d08a8c2e.png)

**Do note the change from remote.host.fdns to remote.host.ip**

The operation ends with no error message after the successful parsing of shares from the ARP table lookup

![image](https://user-images.githubusercontent.com/89908497/184954604-f83ee9cd-ec59-4281-8c71-c169064c3ddc.png)

This is my first operation so please let me know what I've done wrong, I would love to learn.

**Comment by clenk** [comment]
Hi, at the bottom of the json report there's a section for "skipped_abilities" that shows that the agent became untrusted so the last 2 abilities were not run on it. I would check if something happened to cause the agent to not respond.

Two of your screenshots are identical so I'm not sure what the commands were changed to.

With regards to the abilities, abilities can use facts in their commands, but can also have requirements, which are facts that must be present in order to run. Abilties can also have parsers, which parse the output of their command into new facts. If you view the ability in the abilities tab you can see its requirements and parsers. If you want to remove a requirement click on the "`[-]`":
![remove_requirement](https://user-images.githubusercontent.com/402940/184983725-469af409-649e-4bdd-bd5d-90357e4aaba4.PNG)

When you edit the ability to use IPs you'll want to make sure the parsers in the previous ability use the same fact name as in the command and requirement of the next one.

**Comment by bcheevers123** [comment]
@clenk Sorry for the duplicate image here is the correct one:

![image](https://user-images.githubusercontent.com/89908497/185096856-de40f081-81b4-46d4-be83-896f47001e96.png)


I tried to edit the ability to remove thee parser and requirement. It now no longer reports the need for the remote.host.fqdn.

Ran the operation again, and I now get this error on

![image](https://user-images.githubusercontent.com/89908497/185351187-48cefe6b-2d38-4da7-b4d7-823952dced4f.png)

on this command

![image](https://user-images.githubusercontent.com/89908497/185351323-0ecba555-4be7-4d6b-b225-5917f6065d95.png)

I do also see that caldera now tries to view the shares of every ipaddress found in the ARP table, including the already hacked Windows machine. This gets a little further, with a successful copying of the sandcat, but doesnt bother with last command due to agent becoming untrusted (I am not sure why this happens)

![image](https://user-images.githubusercontent.com/89908497/185351863-cf121618-d568-4d76-80e1-b84d80b3c943.png)


I have attached the full report.



[Wormz_full-report.txt](https://github.com/mitre/caldera/files/9371952/Wormz_full-report.txt)

**Comment by clenk** [comment]
The "Access is denied" error suggests that maybe the agent needs to be run as administrator.

One thing you could try for the agent going untrusted is to modify the agent settings (see [documentation](https://caldera.readthedocs.io/en/latest/Basic-Usage.html#agent-settings)), like increasing the untrusted timer.

**Comment by bcheevers123** [comment]
I have tried this and the result is still the same. Is their a step by step walkthrough of setting up a WIndows Worm scenario?

**Comment by clenk** [comment]
No, not at this time. Are you able to run other adversary profiles, such as the `Check` adversary, or do they also fail with "agent untrusted" in the "skipped_abilities" section of the json report?

**Comment by bcheevers123** [comment]
I have ran the check advisory command and I get this output. Looks like it went to completion and failed the last task due to no executor
[Check_full-report.txt](https://github.com/mitre/caldera/files/9395605/Check_full-report.txt)
r
